packages/EasyCodingStandard/packages/ChangedFilesDetector/src/Cache/CacheFactory.php

```diff
@@ -18,11 +29,16 @@ public function create(): Cache
 
     private function getCacheDirectory(): string
     {
+        $cacheDirectory = $this->parameterProvider->provideParameter('cache_directory');
+        if ($cacheDirectory === null) {
+            $cacheDirectory = sys_get_temp_dir() . DIRECTORY_SEPARATOR . '_changed_files_detector';
+        }
```

Comment:
Using default values with null check could be messy to debug.

I wonder if this `$cacheDirectory` could be provided by constructor.
And default value set in Kernel. Same as `$appKernel`, do you know what I mean?

I'll explain bit more, if you need.

Overall I like this service based approach :+1: